main.go

```diff
@@ -163,11 +165,16 @@ func main() {
 		setupLog.Info("setting up operator to watch local namespace")
 		mgrOptions.Namespace = oprNamespace
 	} else {
-		mgrOptions.Namespace = ""
 		if watchList != nil {
-			setupLog.Info("setting up operator to watch multiple namespaces", "namespace(s)", watchList)
-			mgrOptions.NewCache = cache.MultiNamespacedCacheBuilder(watchList)
+			if len(watchList) == 1 {
+				setupLog.Info("setting up operator to watch single namespace")
+				mgrOptions.Namespace = watchList[0]
+			} else {
+				setupLog.Info("setting up operator to watch multiple namespaces", "namespace(s)", watchList)
+				mgrOptions.NewCache = cache.MultiNamespacedCacheBuilder(watchList)
+			}
```

Comment:
Is this change related to the cert-manager?

Reply:
I tried to revert this change and I didn't get compile errors, could you double check please?

Reply:
Yes you can compile but you will get runtime error when you run tests.

Reply:
I see, since controller-runtime v0.15.0 `must specify more than one namespace to use multi-namespace cache`
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.15.0/pkg/cache/multi_namespace_cache.go#L54